### PR TITLE
add hasProfession check

### PR DESCRIPTION
**Pull request by adrsch04**
### What?
A check for the player's profession

### Why?
It's needed for some sections.

### How?
It was implemented the same way as all other checks like hasPossession.

### CODE REVIEW

**Comment by Nilstrieb** [comment]
is this ready for review?

File: fabbwled-backend/section-schema.json [hunk 1]
```diff
@@ -26,7 +26,7 @@
             "checkTickBox": {
               "description": "Checks the tickbox",
               "type": "boolean"
-            }
+            },
```
Comment (Nilstrieb) [inline fabbwled-backend/section-schema.json:29]: oh no, we really need to generate the schema, or at least validate it, see #506

**Review by Lion-alt — APPROVED** [review]
Apart from the syntax errors in section-schema.json LGTM 👍

File: fabbwled-backend/src/main/java/ch/bbw/fabbwled/lands/book/yaml/Condition.java [hunk 2]
```diff
@@ -42,4 +45,7 @@ public boolean isActive(PlayerSession session, SectionId section) {
         }
     }
 
+
+    record HasProfession(ProfessionEnum profession) implements Condition {}
```
Comment (Nilstrieb) [inline fabbwled-backend/src/main/java/ch/bbw/fabbwled/lands/book/yaml/Condition.java:49]: please implement `isActive`

**Review by lucasmillanm — APPROVED** [review]

**Review by dbabvwid — APPROVED** [review]

**Review by elion004 — APPROVED** [review]
LGTM

**Review by corsin-lehre — APPROVED** [review]
Nils told me its good

**Review by Nilstrieb — APPROVED** [review]

**Pull request merged**